web/js/MapUI/Components/UpdateProjection/UpdateProjection.js

```diff
@@ -0,0 +1,399 @@
+        preloadForCompareMode();
+        return;
+      case layerConstants.TOGGLE_OVERLAY_GROUPS:
+        return reloadLayers();
+      case paletteConstants.SET_THRESHOLD_RANGE_AND_SQUASH:
+      case paletteConstants.SET_CUSTOM:
+      case paletteConstants.SET_DISABLED_CLASSIFICATION:
+      case paletteConstants.CLEAR_CUSTOM:
+      case layerConstants.ADD_LAYERS_FOR_EVENT:
+        return setTimeout(reloadLayers, 100);
+      case vectorStyleConstants.SET_FILTER_RANGE:
+      case vectorStyleConstants.SET_VECTORSTYLE:
+      case vectorStyleConstants.CLEAR_VECTORSTYLE:
+      case SET_SCREEN_INFO:
+        return onResize();
+      default:
+        break;
+    }
+  };
+
+  const flyToNewExtent = function(extent, rotation) {
+    const coordinateX = extent[0] + (extent[2] - extent[0]) / 2;
+    const coordinateY = extent[1] + (extent[3] - extent[1]) / 2;
+    const coordinates = [coordinateX, coordinateY];
+    const resolution = ui.selected.getView().getResolutionForExtent(extent);
+    const zoom = ui.selected.getView().getZoomForResolution(resolution);
+    // Animate to extent, zoom & rotate:
+    // Don't animate when an event is selected (Event selection already animates)
+    return fly(ui.selected, proj, coordinates, zoom, rotation);
+  };
+
+  /**
+ * Create a Layergroup given the date and layerGroups
+ */
+  async function getCompareLayerGroup([compareActiveString, compareDateString], state, granuleOptions) {
+    const { createLayer } = ui;
+    const compareSideLayers = getActiveLayers(state, compareActiveString);
+    const layers = getLayers(state, { reverse: true }, compareSideLayers)
+      .map(async (def) => {
+        const options = {
+          ...getGranuleOptions(state, def, compareActiveString, granuleOptions),
+          date: getSelectedDate(dateCompareState, compareDateString),
+          group: compareActiveString,
+        };
+        return createLayer(def, options);
+      });
+    const compareLayerGroup = await Promise.all(layers);
+
+    return new OlLayerGroup({
+      layers: compareLayerGroup,
+      date: getSelectedDate(dateCompareState, compareDateString),
+      group: compareActiveString,
+    });
+  }
+
+  /**
+  * Remove Layers from map
+  *
+  * @method clearLayers
+  * @static
+  *
+  * @param {object} map - Openlayers Map obj
+  *
+  * @returns {void}
+  */
+  const clearLayers = function() {
+    const activeLayers = ui.selected
+      .getLayers()
+      .getArray()
+      .slice(0);
+    lodashEach(activeLayers, (mapLayer) => {
+      ui.selected.removeLayer(mapLayer);
+    });
+    ui.cache.clear();
+  };
+
+  /**
+   * @method reloadLayers
+   *
+   * @param {object} map - Openlayers Map obj
+   * @param {Object} granuleOptions (optional: only used for granule layers)
+   *   @param {Boolean} granuleDates - array of granule dates
+   *   @param {Boolean} id - layer id
+   * @returns {void}
+   */
+  async function reloadLayers(granuleOptions) {
+    const map = ui.selected;
```

Comment:
"map" is already declared on line 42.

Reply:
This variable represents something slightly different than what was declared in props on line 42 so I have renamed the variable within the `reloadLayers` function.